tests/unit/exception_logging_test.cc

```diff
@@ -129,6 +129,39 @@ BOOST_AUTO_TEST_CASE(nested_exception_logging2) {
     BOOST_REQUIRE_EQUAL(log_msg.str(), std::string("std::nested_exception: <no exception>"));
 }
 
+class very_important_exception : public std::exception {
+    const char* my_name = "very important information";
+
+public:
+    const char* what() const noexcept {
+        return my_name;
+    }
+};
+
+// Test logging of nested exception that have std::system_error mixed with other exceptions
+// so that std::system_error is in the middle of the exception chain.
+BOOST_AUTO_TEST_CASE(nested_exception_logging3) {
+    std::ostringstream log_msg;
+
+    try {
+        throw very_important_exception();
+        } catch (...) {
+        try {
+            std::throw_with_nested(std::system_error(1, std::generic_category(), "my error"));
+        } catch (...) {
+            try {
+                std::throw_with_nested(unknown_obj("This is an unknown object"));
+            } catch (...) {
+                log_msg << std::current_exception();
+            }
+        }
+    }
+
+    std::string expected_string("std::_Nested_exception<unknown_obj>: std::_Nested_exception<std::system_error> (error generic:1, my error: Operation not permitted): very_important_exception (very important information)");
```

Comment:
Unrelated to this patch, but an observation related to my previous comments to you and @eliransin -
I said it might have been mistake to have a special case for std::system_error, because this already has a what(). And we can see here that indeed it wasn't needed.... Adding the code() just added the message "error generic:1", but that is unndeeded - what() already printed it as "Operation not permitted", which is the textual representation of this "1". Nobody needs to see this "1".

Reply:
@nyh The intention of this patch was not to change special casing that pre-existed.
However to answer your question: I guess that it could be that not all possible values have a string representation. And in that case some "unknown error code" string would be returned by `what()`. In such a case you want to see the numeric value.

Reply:
@nyh Is there anything that prevents from merging this patch? We really need this to get moving.

Reply:
@eliransin

Reply:
No, I'm testing it now...

Reply:
> @nyh The intention of this patch was not to change special casing that pre-existed.
> However to answer your question: I guess that it could be that not all possible values have a string representation. And in that case some "unknown error code" string would be returned by `what()`. In such a case you want to see the numeric value.

I argue that in this case, it is the duty of the error's `what()` to print the number and not some "unknown error". But as I said, it's not related to this patch. I'm just continuing an earlier discussion.